drivers/gpio/gpio_sedi.c

```diff
@@ -0,0 +1,329 @@
+	sedi_gpio_t gpio_dev = config->device;
+	sedi_gpio_pin_config_t pin_config = { 0 };
+
+	/* Only input needs interrupt enabled */
+	pin_config.direction = SEDI_GPIO_DIR_MODE_INPUT;
+	pin_config.enable_wakeup = true;
+	if (mode == GPIO_INT_MODE_DISABLED) {
+		pin_config.enable_interrupt = false;
+	} else {
+		pin_config.enable_interrupt = true;
+		if (mode == GPIO_INT_MODE_LEVEL) {
+			return -1;
+		}
+		switch (trig) {
+		case GPIO_INT_TRIG_LOW:
+			pin_config.interrupt_mode =
+				SEDI_GPIO_INT_MODE_FALLING_EDGE;
+			break;
+		case GPIO_INT_TRIG_HIGH:
+			pin_config.interrupt_mode =
+				SEDI_GPIO_INT_MODE_RISING_EDGE;
+			break;
+		case GPIO_INT_TRIG_BOTH:
+			pin_config.interrupt_mode =
+				SEDI_GPIO_INT_MODE_BOTH_EDGE;
+			break;
+		}
+	}
+	/* Configure interrupt mode */
+	sedi_gpio_config_pin(gpio_dev, pin, pin_config);
+
+	return 0;
+}
+
+static int gpio_sedi_manage_callback(const struct device *dev,
+				     struct gpio_callback *callback,
+				     bool set)
+{
+	struct gpio_sedi_data *data = dev->data;
+
+	gpio_manage_callback(&(data->callbacks), callback, set);
+
+	return 0;
+}
+
+static uint32_t gpio_sedi_get_pending(const struct device *dev)
+{
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+
+	return sedi_gpio_get_gisr(gpio_dev, 0);
+}
+
+static const struct gpio_driver_api gpio_sedi_driver_api = {
+	.pin_configure = gpio_sedi_configure,
+	.port_get_raw = gpio_sedi_get_raw,
+	.port_set_masked_raw = gpio_sedi_set_masked_raw,
+	.port_set_bits_raw = gpio_sedi_set_bits_raw,
+	.port_clear_bits_raw = gpio_sedi_clear_bits_raw,
+	.port_toggle_bits = gpio_sedi_toggle_bits,
+	.pin_interrupt_configure = gpio_sedi_interrupt_configure,
+	.manage_callback = gpio_sedi_manage_callback,
+	.get_pending_int = gpio_sedi_get_pending
+};
+
+extern void gpio_isr(IN sedi_gpio_t gpio_device);
+
+static int gpio_sedi_init(const struct device *dev)
+{
+	int ret;
+	const struct gpio_sedi_config *config = dev->config;
+	sedi_gpio_t gpio_dev = config->device;
+
+	/* Call sedi gpio init */
+	ret = sedi_gpio_init(gpio_dev, gpio_sedi_callback, (void *)dev);
+
+	if (ret != 0) {
+		return ret;
+	}
+	sedi_gpio_set_power(gpio_dev, SEDI_POWER_FULL);
+
+	config->irq_config();
+
+	return 0;
+}
+
+#define GPIO_SEDI_IRQ_FLAGS_SENSE0(n) 0
+#define GPIO_SEDI_IRQ_FLAGS_SENSE1(n) DT_INST_IRQ(n, sense)
+#define GPIO_SEDI_IRQ_FLAGS(n) \
+	_CONCAT(GPIO_SEDI_IRQ_FLAGS_SENSE, DT_INST_IRQ_HAS_CELL(n, sense))(n)
+
+#define GPIO_DEVICE_INIT_SEDI(n)				       \
+	static struct gpio_sedi_data gpio##n##_data;	               \
+	static void gpio_sedi_irq_config_##n(void)		       \
+	{							       \
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority), \
+			    gpio_isr, n,			       \
+			    GPIO_SEDI_IRQ_FLAGS(n));		       \
+		irq_enable(DT_INST_IRQN(n));			       \
+	};							       \
+	static const struct gpio_sedi_config gpio##n##_config = {      \
+		.common = { 0xFFFFFFFF },			       \
+		.device = SEDI_GPIO_##n,			       \
```

Comment:
add peripheral_id for SEDI instance, see sedi_i2c.c.
no sequence guarantee from DT FOREACH, considering there might be more GPIO devices in future ISH SoCs.

Reply:
done